haystack/utils/deepsetcloud.py

```diff
@@ -635,6 +637,79 @@ def _build_workspace_url(self, workspace: Optional[str] = None):
+    def __init__(self, client: DeepsetCloudClient, workspace: Optional[str] = None, index: Optional[str] = None):
+        """
+        A client to communicate with Deepset Cloud evaluation sets and labels.
+
+        :param client: Deepset Cloud client
+        :param workspace: workspace in Deepset Cloud
+
+        """
+        self.client = client
+        self.workspace = workspace
+        self.index = index
+
+    def get_labels(self, index: str, workspace: Optional[str] = None) -> List[Label]:
+        evaluation_set = self._get_evaluation_set(index=index, workspace=workspace)[0]
+
+        labels = self._get_labels_from_evaluation_set(
+            workspace=workspace, evaluation_set_id=evaluation_set["evaluation_set_id"]
+        )
+
+        return [
+            Label(
+                query=label_dict["query"],
+                document=Document(content=label_dict["context"]),
+                is_correct_answer=True,
+                is_correct_document=True,
+                origin="user-feedback",
+                answer=label_dict["answer"],
+                id=label_dict["label_id"],
+                no_answer=True if label_dict.get("answer", None) else False,
+                pipeline_id=None,
+                created_at=label_dict["created_at"],
+                updated_at=label_dict["updated_at"],
+                meta=label_dict["meta"],
+                filters={},
+            )
+            for label_dict in labels
+        ]
+
+    def get_labels_count(self, index: Optional[str] = None, workspace: Optional[str] = None) -> int:
+        evaluation_set = self._get_evaluation_set(index=index, workspace=workspace)
+        return evaluation_set[0]["total_labels"]
+
+    def get_evaluation_set_names(self, workspace: Optional[str] = None):
+        evaluation_sets_response = self._get_evaluation_set(index=None, workspace=workspace)
+
+        return [eval_set["name"] for eval_set in evaluation_sets_response]
+
+    def _get_evaluation_set(self, index: Optional[str], workspace: Optional[str] = None) -> List[dict]:
+        if not index:
+            index = self.index
+
+        url = self._build_workspace_url(workspace=workspace)
+        evaluation_set_url = f"{url}/evaluation_sets"
+        evaluation_sets_response = self.client.get(url=evaluation_set_url, query_params={"name": index}).json()
+
+        return evaluation_sets_response.get("data", [])
+
+    def _get_labels_from_evaluation_set(
+        self, workspace: Optional[str] = None, evaluation_set_id: Optional[str] = None
+    ) -> Generator[dict]:
+        url = f"{self._build_workspace_url(workspace=workspace)}/evaluation_sets/{evaluation_set_id}"
+        labels = self.client.get(url=url).json()
+
+        for label in labels:
+            yield label
+
+    def _build_workspace_url(self, workspace: Optional[str] = None):
+        if workspace is None:
+            workspace = self.workspace
+        return self.client.build_workspace_url(workspace)
```

Comment:
It's not much that `EvaluationSetClient` and `PipelineClient` share, but maybe it's worth a common base class `WorkspaceClient` in order to avoid duplicate code.

Reply:
Should we handle it in this PR or another one?

For now it isn't much, you are right, but it could grow in the future. We could also put all request functionality in this class and let `EvaluationSetClient` and `PipelineClient only serialize data

Reply:
Yep, we can do that in another PR.